portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,72 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.*;
+import com.google.gson.Gson;
 import java.io.IOException;
+import javax.servlet.http.*;
 import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet(urlPatterns = "/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+    String movie = request.getParameter("movie");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    // Filter movieFilter = new FilterPredicate("movie", FilterOperator.EQUAL, movie);
+    // Query query = new Query("Comment").setFilter(movieFilter).addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query(movie).addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery pq = datastore.prepare(query);
+    List<Entity> resultsList = pq.asList(FetchOptions.Builder.withLimit(maxComments));
+
+    ArrayList<String> comments = new ArrayList<>();
+    for (Entity entity : resultsList) {
+        String text = (String) entity.getProperty("text");
+
+        comments.add(text);
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJSON(comments));
+  }
+
+  private String convertToJSON(ArrayList<String> arr) {
+    Gson gson = new Gson();
+    String json = gson.toJson(arr);
+    
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long timestamp = System.currentTimeMillis();
+    String comment = this.getComment(request);
+    String movie = request.getParameter("curr-movie");
+
+    // Entity commentEntity = new Entity("Comment");
+    Entity commentEntity = new Entity(movie);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("text", comment);
+    // commentEntity.setProperty("movie", movie);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/movies.html");
+  }
+
+  private String getComment(HttpServletRequest request) {
+    String comment = request.getParameter("comment");
```

Comment:
Do you need a new helper method for this? Will there be more complicated logic added later?